entity.module

```diff
@@ -86,3 +89,54 @@ function entity_module_preuninstall($module) {
     \Drupal::service('entity.bundle_plugin_installer')->uninstallBundles($entity_type, [$module]);
   }
 }
+
+/**
+ * Implements hook_query_TAG_alter().
+ *
+ * Performs access filtering for entity queries.
+ */
+function entity_query_entity_query_alter(Select $query) {
+  $entity_type_id = $query->getMetaData('entity_type');
+  if ($query->hasTag($entity_type_id . '_access')) {
+    $entity_type_manager = \Drupal::entityTypeManager();
+    $entity_type = $entity_type_manager->getDefinition($entity_type_id);
+    SqlQueryAlter::create(\Drupal::getContainer())->queryAlter($query, $entity_type);
+  }
+}
+
+/**
+ * Implements hook_query_TAG_alter().
+ *
+ * Performs access filtering for views queries.
+ */
+function entity_query_views_alter(Select $select) {
+  $tables_info = $select->getTables();
+  $table_info = reset($tables_info);
+  if ($entity_type = _entity_get_entity_type_id_from_table($table_info['table'])) {
+    SqlQueryAlter::create(\Drupal::getContainer())->queryAlter($select, $entity_type);
+  }
+}
+
+/**
+ * Determine the entity type of a query based upon the table name.
+ *
+ * @param string $table_name
+ *   The table name.
+ *
+ * @return \Drupal\Core\Entity\EntityTypeInterface|null
+ *   The entity type, or NULL if unknown.
+ */
+function _entity_get_entity_type_id_from_table($table_name) {
+  foreach (\Drupal::entityTypeManager()->getDefinitions() as $entity_type) {
+    $tables = [
+      $entity_type->getBaseTable(),
+      $entity_type->getDataTable(),
+      $entity_type->getRevisionTable(),
+      $entity_type->getRevisionDataTable(),
+    ];
```

Comment:
Building this map could be statically cached and moved to a helper method/function like `getEntityTypeTableNames()`, I'm not sure if that'd be worth it.